traits/metricstrait/README.md

```diff
@@ -1,26 +1,45 @@
 # OAM Metrics Trait
 This trait is designed to provide an easy way for an OAM application operator to collect metrics from any workload
- that emits metrics through an endpoint reachable from within the cluster. The metrics traits relies on [Prometheus operator]() and
- [Prometheus helm chart]() 
- 
-## Collect metrics through an existing service
+that emits metrics through an endpoint reachable from within the cluster. 
+The metrics traits relies on [Prometheus operator](https://github.com/coreos/prometheus-operator) and
+[Prometheus helm chart](https://github.com/helm/charts/tree/master/stable/prometheus-operator). 
+
+With this trait, an application operator only needs to specify a few fields to extract metrics from the workloads the trait attached to.
+Here are the brief descriptions of each field and its usage.
 
+| Name | Optional | Default | Description |
+| ---- | ----------- | -------- | --------- |
+| format            | Yes | "prometheus" | The format of the data. The only format we support now is prometheus|
+| portName          | Yes | N/A |  The name of the port in a service that exposes the workload. This is mutually exclusive with targetPort/targetSelector |
+| targetPort        | Yes | N/A  | The pod port that exposes the metrics data. This is mutually exclusive with portName.  |
+| targetSelector    | Yes  | N/A | The selector that can be used to label select target pods. The default uses the labels on the workload.|
+| path              | Yes  | "/metrics" | path to scrape for metrics from the end point.  |
+| scheme            | Yes  | "http"  | The scrap scheme that we support. We only support http for now.  |
+| enabled           | Yes  | "true"  | Switch for the trait to be turned on or off. |
```

Comment:
Sorry for late comments, my 2c regarding to this spec:

1. Fields with default values should be skipped in readme and other samples unless different values are given.
2. I personally think `portName` and `targetPort + targetSelector` design is still complicated.

Allow me to highlight (2) a bit. For a monitoring trait, it only need to assume to scrape metrics from given exposed ports, it should not certain with certain `Service` exists or not. Actually, there's even no such concept of `Service` in OAM (which is by design).

Thus, I think a full spec like below should work?

```
 apiVersion: standard.oam.dev/v1alpha1
 kind: Monitoring
 spec:
   # should be a list
   scrapeList:
   - targetPort: 8080
     path: "/foo"
     scheme:  "https"
     format: "baz"
     active: false # active seems better than enabled?
```
Note that `portName` or `targetSelector` is not needed, because OAM should assume `8080` as an exposed port (i.e. applied an Expose trait exists with `8080:[container_port]`). If not, this OAM runtime should automatically apply the expose trait (i.e. an Expose trait with `8080:8080`)

Reply:
FYI, design of Expose trait in my mind: https://github.com/oam-dev/catalog/issues/53

Reply:
We are not that far. 
1. The portName is for the case that workload controller already creates a service (like containerizedWorkload). This is also very OAM native as it makes use the "childResrouces" field in the workloadDefinition and the whole trait/workload interaction mechanism designed for OAM.
2. The selector is just for the case that the pod label is not the same as the workload.
3. I think the list style is pretty different from the current CSE style but I am ok if they want to move to that direction in the future.

Reply:
As we discussed in the weekly meeting, I am trying to strike a balance between absolutely minimal (which in this case is an empty trait, everything has a relatively good default value) and some useful configurations. However, obviously that the current design is purely subjective and opinionated.

Reply:
@ryanzhang-oss For (1), Not sure why user can't just fill in the ContainerizedWorkload's `containerPort` as `targetPort` here (as he already know that port is exposed)?

For (3), the motivation is I'd prefer a "list" than "multiple traits with same kind" in AppConfig. WDYT? Btw, we don't need to force CSE move to this spec, the goal is this spec can be used with ARMS Prometheus seamlessly.